### PR TITLE
Add a method for debug messages

### LINKED ISSUE #5: Debug messages
I think it would be helpful to have a function that logs a message only if gd.debug, instead of having to do separate if statements.

### PR DESCRIPTION
**Pull request by riftEmber**
Allows logging messages only if a specific debug category is enabled
Fixes #5

### CODE REVIEW

**Comment by riftEmber** [comment]
I don't know how we want the command to look for enabling or disabling various categories

File: index.js [hunk 1]
```diff
@@ -346,6 +346,19 @@ var distance = function() {
 	}
 };
 
+// log a debug message only if its category is enabled
+// first argument is category, following arguments are console.log's input
+var debugMsg = function() {
+	var category = arguments[0];
+	if (category in gd.debug) {
```
Comment (vpzomtrrfrt) [inline index.js]: Codacy Issue found: ["gd" is not defined.](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork-server/file/2647095345/issues/source?bid=3248399&fileBranchId=3323017#l353)

Comment (riftEmber) [inline index.js]: oops

File: index.js [hunk 2]
```diff
@@ -625,6 +625,19 @@ var commands = {
...
```
Comment (vpzomtrrfrt) [inline index.js]: Codacy Issue found: ["gd" is not defined.](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork-server/file/2647138125/issues/source?bid=3248399&fileBranchId=3323017#l632)

**Comment by riftEmber** [comment]
Oh wait, that won't work. I guess I have to pass in gd

**Comment by riftEmber** [comment]
Ok we now have one debug category, "winning"

**Comment by vpzomtrrfrt** [comment]
```
/home/colin/turbo-spork-server/index.js:610
                        if (category in games[room].debug) {
                                                   ^

TypeError: Cannot use 'in' operator to search for 'hi' in undefined
    at Object.commands.chat.handler (/home/colin/turbo-spork-server/index.js:610:34)
    at WebSocket.handleMessage (/home/colin/turbo-spork-server/index.js:678:9)
    at emitTwo (events.js:106:13)
    at WebSocket.emit (events.js:191:7)
    at Receiver.ontext (/home/colin/turbo-spork-server/node_modules/ws/lib/WebSocket.js:841:10)
    at /home/colin/turbo-spork-server/node_modules/ws/lib/Receiver.js:536:18
    at Receiver.applyExtensions (/home/colin/turbo-spork-server/node_modules/ws/lib/Receiver.js:371:5)
    at /home/colin/turbo-spork-server/node_modules/ws/lib/Receiver.js:508:14
    at Receiver.flush (/home/colin/turbo-spork-server/node_modules/ws/lib/Receiver.js:347:3)
    at Receiver.opcodes.1.finish (/home/colin/turbo-spork-server/node_modules/ws/lib/Receiver.js:541:12)
```

**Comment by riftEmber** [comment]
Should be fixed now

**Comment by riftEmber** [comment]
I guess I'll also add some more debug messages

**Comment by riftEmber** [comment]
Actually let's just add them when we need them. Think this is ready to merge?

**Comment by vpzomtrrfrt** [comment]
If it works

**Comment by vpzomtrrfrt** [comment]
which it doesn't
debugoff

**Comment by riftEmber** [comment]
It works now, I'll merge it

**Pull request merged**